userspace/engine/rule_loader.cpp

```diff
@@ -21,7 +21,12 @@ limitations under the License.
 #include "filter_evttype_resolver.h"
 #include "filter_warning_resolver.h"
 #include <version.h>
-#include <sstream>
+
+#ifndef _WIN32
```

Comment:
This is not enough and that's also the reason why the CI builds are failing. We have a very weird setup in the libraries for regex support that depends on:
- old compiler versions, that don't support `<regex>`
- inclusion or not of `oniguruma` in the build, which is a dependency of `jq`

To solve all the combinations, this trick worked for me:
- https://github.com/falcosecurity/libs/blob/d08daff1bf69c2388b7474c1fed96bd6e98b900f/userspace/libsinsp/filter/parser.cpp#L35
- https://github.com/falcosecurity/libs/blob/d08daff1bf69c2388b7474c1fed96bd6e98b900f/userspace/libsinsp/CMakeLists.txt#L272 (I think you should be able to use the `USE_BUNDLED_ONIGURUMA` definition from libsinsp's cmake file)

Mind that this would also cause the regex standard used in the code to differ, depending on what regex library we are building with. So you need to write two copies of each regex, and also handle them in two different ways:
- https://github.com/falcosecurity/libs/blob/d08daff1bf69c2388b7474c1fed96bd6e98b900f/userspace/libsinsp/filter/parser.cpp#L46
- https://github.com/falcosecurity/libs/blob/d08daff1bf69c2388b7474c1fed96bd6e98b900f/userspace/libsinsp/filter/parser.cpp#L532

Good news is that the CI and the tests won't pass until this is addressed: we have different build types that help us test all the cases.

Bad news is that this is truly terrible, and I really hope we'll be able to do better than this as soon as possible. The bottleneck is the presence of `oniguruma` brought in by `jq`, we we manage to remove that we should be able to just stick to one regex standard (posix I guess).

Reply:
dropping fix for the list inserter altogether - I don't have resources to fight regex portability here.
@jasondellaluce keep in mind that the current list expansion routine is quite inefficient - it keeps on looking for all available lists (100s) in the continuously growing string and does it from the string beginning each time.

Reply:
Unfortunately, that's what we had in the past with Lua as well. This part will need to be substituted with proper AST traversal+substitutions in the future, however that would introduce ruleset breaking changes so it's something that we'll tackle in the future.